staging/src/k8s.io/apimachinery/pkg/api/operation/operation.go

```diff
@@ -49,16 +49,27 @@ type Operation struct {
 
 // Request provides information about the request being validated.
 type Request struct {
-	// Subresources is the path to the subresource being validated. Validations
-	// should use this only when the validation of field differs for a particular
-	// subresource. For example, the resize subresource of pod is allowed to change
-	// the value of spec.container[*].resources fields, which are immutable updated
-	// via the root pod resource. In this example, because validation of the fields
-	// are conditional to the subresource, the validator should use Subresources.
-	// Note field wiping, which limits which fields a subresource is allowed to write
-	// to, is handled in strategies, and does should to be implemented using
-	// Subresources since field wiping drops field changes from the request, but does
-	// not require changes to validation logic.
+	// Subresources identifies the subresource path components of the request. For
+	// example, Subresources for a request to `/api/v1/pods/my-pod/status` would be
+	// `["status"]`. For `/api/v1/widget/my-widget/x/y/z`, it would be `["x", "y",
+	// "z"]`. For a root resource (`/api/v1/pods/my-pod`), Subresources will be an
+	// empty slice.
+	//
+	// Validation logic should only consult this field if the validation rules for a
```

Comment:
What I still don't understand is HOW to use this.

In the example of pod resources, we find:

```
    // Compute Resources required by this container.
    // Cannot be updated.
    // More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
    // +optional
    Resources ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,8,opt,name=resources"`
```

How do you think we will express `immutable unless subresources == ["scale"]`

Reply:
I'll have to answer this in the validation tag implementations.   I'm hoping this folds into something like `+k8s:if("subresources == ["scale"])`=<subresource specific validation>` so we don't need special tags (like we were hoping to do for options).